src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -57,13 +57,11 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
-                _trace.Error(errorText);
-
-                throw new InvalidOperationException(errorText);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid directory location.", _folderSource.Path);
+                var ex = new InvalidOperationException(errorText);
+                _trace.Error(errorText, ex);
```

Comment:
The idea was the user will get an error to fix the path but the host will start in case there are other functions/triggers. currently nothing gets started and all other functions/triggers in the function app will get blocked. no I can't validate the path any earlier than this. this is called when the json file gets saved or read after the function app starts.